atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/async/CqlClientImpl.java

```diff
@@ -116,7 +121,7 @@ public void close() {
             this.rowStreamAccumulator = rowStreamAccumulator;
         }
 
-        final AsyncFunction<ResultSet, R> iterate() {
+        final AsyncFunction<ResultSet, R> iterate(Executor executor) {
```

Comment:
From reading the code, I believe it's because you want to do transforms but outside the client but using the CqlClient's executor so you introduce another deferred wrapper around it?

It feels like you have two logical pieces here, 

* singular gets
* something that takes a batch, spreads it, and then contracts again

the latter feels like it has nothing to do with cql at this point, so if you were to have some sort of `AsyncCellLoader` or some better name, that had its own executor, then you can schedule the combining on that thread perhaps? agree disagree @jeremyk-91 ?